tests/main_test.py

```diff
@@ -97,8 +103,8 @@ def test_bad_payment(self):
                 'preferred': 'cashapp'
```

Comment:
Add another test case for only updating "preferred" where the value supplied isn't actually allowed.
For example "preferred": "non-existent" should not be successful